dask/array/numpy_compat.py

```diff
@@ -262,3 +263,11 @@ def sliding_window_view(
         return np.lib.stride_tricks.as_strided(
             x, strides=out_strides, shape=out_shape, subok=subok, writeable=writeable
         )
+
+
+# kwarg is renamed in numpy 1.22.0
+def percentile(a, q, method="linear"):
+    if _numpy_122:
+        return np.percentile(a, q, method=method)
+    else:
+        return np.percentile(a, q, interpolation=method)
```

Comment:
@pentschev I think this is what you are suggesting right?

Reply:
Internally, yes, but the Dask function prototype [now only contains `method`], that's the public `da.percentile` now, right? Or am I confused and it's this one from `numpy_compat.py`?

If the first one is the public `da.percentile` I think it will fail to dispatch on older NumPy because there's no `interpolate` kwarg.

Reply:
Nevermind, you're right:

```python
In [1]: import dask.array as da, cupy as cp, numpy as np

In [2]: np.__version__
Out[2]: '1.20.3'

In [3]: d = da.from_array(cp.ones((16,)), chunks=(4,))

In [4]: qs = np.array([0, 50, 100])

In [5]: result = np.percentile(d, qs, interpolation="midpoint")
/datasets/pentschev/src/dask/dask/array/percentile.py:120: FutureWarning: In Dask 2022.1.0, the `interpolation=` argument to percentile was renamed to `method= `
  warnings.warn(
```

Sorry for the noise here.

Reply:
OOHH I missed that you are talking about the calling `np.percentile` on a dask array directly.

Reply:
```python 
In [1]: import numpy as np
   ...: import dask.array as da
   ...: 
   ...: da.percentile(da.from_array([1,2,3,4]), [0,50], method="linear")
   ...: 
   ...: 
Out[1]: dask.array<percentile, shape=(2,), dtype=float64, chunksize=(2,), chunktype=numpy.ndarray>

In [2]: np.percentile(da.from_array([1,2,3,4]), [0,50], method="linear")
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-3-285571e414ff> in <module>
      2 import dask.array as da
      3 
----> 4 np.percentile(da.from_array([1,2,3,4]),[0,50],  method="linear")

<__array_function__ internals> in percentile(*args, **kwargs)

TypeError: _percentile_dispatcher() got an unexpected keyword argument 'method'

Reply:
Looking those over, I am personally pretty fine with that behavior.

Reply:
Indeed, I wasn't clear. Thanks for double-checking on your end! 😄

Reply:
Yeah phew. This stuff is kind of confusing!

Reply:
What version of NumPy was used for https://github.com/dask/dask/pull/8525#discussion_r779132164? Does that mean that our deprecation message will direct users to something that doesn't work for older NumPy versions?

Reply:
Yeah I had an older version of numpy in there. I think it's ok for us to say latest dask works with latest numpy, no warnings, and otherwise it'll still work, but you might get a warning. Does that feel too harsh?